internal/worker/workerSpecificConfig_test.go

```diff
@@ -0,0 +1,74 @@
+// COPYRIGHT 2024 FERMI NATIONAL ACCELERATOR LABORATORY
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+//
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package worker
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSetWorkerRetryValue(t *testing.T) {
+	opt := SetWorkerRetryValue(GetKerberosTicketsWorkerType, 5)
+	c := &Config{}
```

Comment:
Is the purpose of declaring this pointer to the (struct?) so you can pass by reference later? Or is there some other idea behind that here?

Reply:
Good question.  Short answer: yes, because the function we're testing, `SetWorkerRetryValue`, returns a function that in turn accepts a pointer to a `Config`, not `Config` itself.  Its function signature is:

```go
func SetWorkerRetryValue(w WorkerType, value uint) ConfigOption
```

`ConfigOption` is itself a type alias (see internal/worker/configOption.go) for `func(*Config) error` (in Go, functions are types just like ints, strings, etc). So breaking that out, we could write the function signature of `SetWorkerRetryValue` as:

```go
func SetWorkerRetryValue(w WorkerType, value uint) func(*Config) error
```

So that means that the function we get back from `SetWorkerRetryValue` has to operate on a pointer to `Config` (the compiler will error out if you try to pass in a `Config`).  That's why we declare the pointer to the struct.

More detail:

There are three ways we could declare and/or initialize a pointer to a `Config`:

1. `c := new(Config)` (c is a variable of type pointer to a `Config` (`*Config`))
2. `c := &Config{}` (identical to (1))
3. (custom because we have such a function) `c, err := NewConfig(<arguments>)` (c is a variable of type pointer to a `Config` (`*Config`), `err` is an error that can be `nil` if everything went OK, not-`nil` if something went wrong)

I use (2) here because it makes it very clear that we're storing in `c` a pointer to a `Config` that has none of its fields set (in Go, that means that every value of the struct fields will be the zero-value of that type).  But essentially, it's a pointer to a blank `Config`.